### PR TITLE
Rule: Report unsafe call on nullable types

### PR DESCRIPTION
**Pull request by tagantroy**
This PR implemented #172

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCallOnNullableType.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtUnaryExpression
+
+/**
+ * @author Ivan Balaksha
+ */
+class UnsafeCallOnNullableType(config: Config = Config.empty) : Rule(config) {
+    override val issue: Issue = Issue("UnsafeCallOnNullableType", Severity.Defect, "")
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCallOnNullableType.kt]: As with #154 we are trying to have a description for each rule, please write 1-2 sentences why unsafe calls are bad ^^

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCallOnNullableTypeSpec.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.jetbrains.spek.subject.SubjectSpek
+import org.assertj.core.api.Assertions.assertThat
+
+/**
+ * @author Ivan Balaksha
+ */
+class UnsafeCallOnNullableTypeSpec : SubjectSpek<UnsafeCallOnNullableType>({
+    subject { UnsafeCallOnNullableType() }
+    describe("check all variants of safe/unsafe calls on nullable types") {
+        it("unsafe call on nullable type") {
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCallOnNullableTypeSpec.kt]: Please add newlines before and after 'it, describe and subject' statements, a style like in `LateinitUsageSpec` are desirable

**Comment by tagantroy** [comment]
@arturbosch updated. Can you take a look?

**Comment by arturbosch** [comment]
Looks good :+1:

**Pull request merged**